client.go

```diff
@@ -184,7 +196,7 @@ func (c *Client) Do(req *Request) (*http.Response, error) {
 		checkOK, checkErr := c.CheckRetry(resp, err)
 
 		if err != nil {
-			c.Logger.Printf("[ERR] %s %s request failed: %v", req.Method, req.URL, err)
+			c.Logger.Printf("[ERR] %s %s request failed: %v (%T)", req.Method, req.URL, err, err)
```

Comment:
Is this still useful or was it more for debugging? We don't usually log the error type in our other packages, seems in this case the `Temporary()` should cover us enough to make this not useful.